File: CLUMP_Python/CLUMP/examples/Example_Euclidean_3D.py
# Script to run GenerateClump_Euclidean_3D
# 2021 © V. Angelidakis, S. Nadimi, M. Otsubo, S. Utili.

# 2021 MATLAB implementation by V. Angelidakis <v.angelidakis@qub.ac>
# 2024 Translated from MATLAB to Python by A.U. Canbolat <utku.canbolat@fau.de>


from CLUMP import GenerateClump_Euclidean_3D
import sys
import os
sys.path.append('../../')

# Load particle shape from stl file
inputGeom = os.path.join(os.path.dirname(__file__), 'ParticleGeometries/Hexahedron.stl')
N = 21
rMin = 0
div = 102
overlap = 0.6
output = 'EU_Hexahedron.txt'
outputVTK = 'EU_Hexahedron.vtk'
visualise = True

mesh, clump = GenerateClump_Euclidean_3D(inputGeom=inputGeom, N=N, rMin=rMin, div=div, overlap=overlap, output=output, outputVTK=outputVTK, visualise=visualise)

print('Total number of spheres: ' + str(clump.numSpheres))

File: CLUMP_MATLAB/examples/Example_Euclidean_3D.m
%% Script to run GenerateClump_Euclidean_3D
% 2021 © V. Angelidakis, S. Nadimi, M. Otsubo, S. Utili.
clc; clear; close all
addpath(genpath('../functions'))  % Add path to in-house functions

%% Load particle shape from stl file
inputGeom='ParticleGeometries/Hexahedron.stl';
N=21;
rMin=0;
div=102;
overlap=0.6;
output='EU_Hexahedron.txt';
visualise=true;

%% Load particle shape from binary 3D image
% inputGeom='ParticleGeometries/Binary_3D_Cube.mat';
% N=21;
% rMin=0;
% div=102;
% overlap=0.6;
% output='FM_3D_image.txt';
% visualise=true;

tic
[mesh, clump]=GenerateClump_Euclidean_3D( inputGeom, N, rMin, div, overlap, output, visualise ); % true to plot mesh and clump
toc

disp(['Total number of spheres: ', num2str(clump.numSpheres)])
